addons/point_of_sale/static/tests/tours/barcode_scanning_tour.js

```diff
@@ -82,6 +80,16 @@ registry.category("web_tour.tours").add("GS1BarcodeScanningTour", {
         ].flat(),
 });
 
+registry.category("web_tour.tours").add("GS1BarcodeScanningTourFallback", {
+    test: true,
+    steps: () =>
+        [
+            Dialog.confirm("Open session"),
+            ProductScreen.scan_barcode_with_error_check("08431673020126"),
+            Chrome.endTour(),
```

Comment:
```suggestion
            ProductScreen.scan_barcode("08431673020126"),
            // <---- expect a error dialog here.
```

I think it's better to just use the existing scan barcode helper followed by a step that expects an error dialog.

Reply:
Also, make sure that the error dialog is something that is explicitly shown, not a traceback.